### PR TITLE
refactor(v2): add common typing package `@docusaurus/types`

### PR DESCRIPTION
**Pull request by gimdongwoo**
## Motivation

I want to convert every package to TypeScript.
Then, I need to common typing package for sharing typings for every package.
But I'm not sure if the package name fits the rules of Docusaurus.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

No

## Related PRs

#1785

### CODE REVIEW

**Comment by gimdongwoo** [comment]
@endiliey 
I refactor all import from `packages/docusaurus/src/server/types` to `@docusaurus/types`.

File: packages/docusaurus-types/package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  "version": "2.0.0-alpha.24",
+  "description": "Common used typings for Docusaurus packages",
+  "main": "./src/index.js",
+  "types": "./src/index.d.ts",
+  "publishConfig": {
+    "access": "public"
+  },
+  "license": "MIT",
+  "dependencies": {
+    "querystring": "0.2.0"
```
Comment (endiliey) [inline packages/docusaurus-types/package.json]: some missing deps here. eg: webpack typing

File: packages/docusaurus/src/server/presets/__tests__/index.test.ts [hunk 2]
```diff
@@ -86,7 +86,7 @@ describe('loadPresets', () => {
       siteConfig: {
         presets: [[path.join(__dirname, '__fixtures__/preset-bar.js')]],
       },
-    } as LoadContext;
+    } as Partial<LoadContext>;
```
Comment (endiliey) [inline packages/docusaurus/src/server/presets/__tests__/index.test.ts]: good catch

File: packages/docusaurus-types/package.json (hunk 1 above)
Comment (gimdongwoo) [reply packages/docusaurus-types/package.json]: Oh! Thank you!

File: packages/docusaurus/src/server/presets/index.ts [hunk 3]
```diff
@@ -7,15 +7,20 @@
...
+import {
+  LoadContext,
+  PluginConfig,
+  Preset,
+  PresetConfig,
+} from '@docusaurus/types';
 
 export function loadPresets(
-  context: LoadContext,
+  context: Partial<LoadContext>,
```
Comment (endiliey) [inline packages/docusaurus/src/server/presets/index.ts]: does it have to be partial ?

Comment (gimdongwoo) [reply packages/docusaurus/src/server/presets/index.ts]: Yes, `loadPresets` method is needed some of `LoadContext`.

**Pull request merged**